src/odemis/acq/stitching/__init__.py

```diff
@@ -16,6 +16,62 @@
 '''
 from __future__ import division
 
-
 from ._registrar import *
 from ._weaver import *
+
+import copy
+
+
+def register(tiles, method="REGISTER_SHIFT"):
+    """ 
+    tiles (list of DataArray of shape YX or tuples of DataArrays): The tiles to compute the registration. 
+    If it's tuples, the first tile of each tuple is the “main tile”, and the following ones are 
+    dependent tiles.
+    method (REGISTER_*): REGISTER_SHIFT → ShiftRegistrar, REGISTER_IDENTITY → IdentityRegistrar
+    returns:
+        tiles (list of DataArray of shape YX or tuples of DataArrays): The tiles as passed, but with updated 
+        MD_POS metadata
+    """
+    
+    if method == "REGISTER_SHIFT":
+        registrar = ShiftRegistrar()
+        
+    elif method == "REGISTER_IDENTITY":
+        registrar = IdentityRegistrar()
+    
+    else:
+        raise ValueError("Invalid registrar") 
+
+    updatedTiles=[]
+    for i in range(len(tiles)):
+        # Separate tile and dependent_tiles
+        if isinstance(tiles[i],tuple):
+            tile = tiles[i][0]
+            dep_tiles = tiles[i][1:]
+        else:
+            tile = tiles[i]
+            dep_tiles = None
+
+        registrar.addTile(tile,dep_tiles)
+        
+        # Update MD_POS metadata
+        md=copy.deepcopy(tile.metadata)
+        md[model.MD_POS] = registrar.getPositions()[0][i]
+        tileUpd = model.DataArray(tile,md)
+        
+        # Return tuple of positions when dependent tiles are present
+        if isinstance(tiles[i],tuple):
+            tilesNew = [tileUpd]
+            for j in range(len(dep_tiles)):
+                md=copy.deepcopy(dep_tiles[j][i].metadata)
+                md[model.MD_POS] = registrar.getPositions()[1][j][i]
```

Comment:
This ```[j][i]``` is annoying. It feels it would make more sense to have ```[i][j]```. We should check if it would be fine to change the registrar API to return the dependent tiles position in the transposed order. One extra advantage is that it would make it easier to pass (and return) a different number of dependent tiles for each tile.